django/tasks/backends/immediate.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from functools import partial
+from inspect import iscoroutinefunction
+from uuid import uuid4
+
+from asgiref.sync import async_to_sync
+
+from django.db import transaction
+from django.tasks.signals import task_enqueued, task_finished
+from django.tasks.task import ResultStatus, TaskResult
+from django.tasks.utils import exception_to_dict, json_normalize
+from django.utils import timezone
+
+from .base import BaseTaskBackend
+
+logger = logging.getLogger(__name__)
+
+
+class ImmediateBackend(BaseTaskBackend):
+    supports_async_task = True
+
+    def _execute_task(self, task_result):
+        """
+        Execute the task for the given `TaskResult`, mutating it with the outcome
+        """
+        object.__setattr__(task_result, "enqueued_at", timezone.now())
```

Comment:
Just curious what the purpose/benefit is of using `object.__setattr__` over `task_result.enqueued_at = timezone.now()`?

Reply:
`task_result` is a frozen dataclass, so setting the attributes directly doesn't work. I'm considering moving it to a private function to facade some of this away.

Reply:
Ah I see, interesting. Thanks!